src/hbbft_bba.erl

```diff
@@ -149,7 +149,7 @@ handle_msg(_Data, _J, _Msg) ->
 bval(Data=#bba_data{f=F}, Id, V) ->
     %% add to witnesses
     Witness = add_witness(Id, V, Data#bba_data.bval_witness, true),
-    WitnessCount = maps:get({val, V}, Witness, 0),
+    WitnessCount = maps:get({val, V}, Witness, 0) + maps:get({val, V}, Data#bba_data.terminate_witness, 0),
```

Comment:
I have some safety concerns here.  The current implementation allows nodes to effectively vote twice for a value.  I /think/ that this is only safe if the witness tracks the source and only allows one vote per node.

Reply:
Which kind of vote, it is technically valid, I think, for a node to vote for a bval value twice?

Reply:
I guess I was thinking that a malign set of nodes could emit both immediately, which would push the node over the threshold I think?

Reply:
add_witness has an 'allow union' flag that prevents changing your vote in the case of a TERM message. BVAL votes are allowed to be made for both values (as the node will emit a {BVAL, b} message once it sees f+1 {BVAL, b} messages).

I'm not clear how many AUX messages can be emitted per round. I *think* just one, but I'm not sure. Currently it allows a node to propose both.

CONF messages don't use add_witness, but they probably should. and they should allow only a single vote from a node per round.

Reply:
I have added a commit to prevent more than one AUX or CONF message per round. TERM was already limited and BVAL should allow multiples.